taichi/system/traceback.cpp

```diff
@@ -6,25 +6,211 @@
 #include "taichi/common/util.h"
 #include <vector>
 #include <iostream>
+#include <string>
+#include <sstream>
+#include <cstdio>
+#include <algorithm>
+#include <memory>
+#include <mutex>
 #include "spdlog/fmt/bundled/color.h"
 
 #ifdef __APPLE__
 #include <execinfo.h>
 #include <cxxabi.h>
 #endif
+#ifdef _WIN64
+#include <intrin.h>
+#include <dbghelp.h>
+
+#include "taichi/platform/windows/windows.h"
+
+#pragma comment(lib, "dbghelp.lib")
+//  https://gist.github.com/rioki/85ca8295d51a5e0b7c56e5005b0ba8b4
+//
+//  Debug Helpers
+//
+// Copyright (c) 2015 - 2017 Sean Farrell <sean.farrell@rioki.org>
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+
+namespace dbg {
+inline void trace(const char *msg, ...) {
```

Comment:
These `inline`s here are no longer necessary since now we are using them in a `cpp`.